From 9108b889c1c9bb3885e962f3b215c3cd7aca86b8 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 24 Feb 2022 12:46:41 +0000
Subject: [PATCH 0982/1000] MANTIS 62298 Mantis 62298: 3rd party dispatch
 monitoring - ConveyorDev.idl vers 1.21 - New attributes dispatchReference,
 dispatchOperationMode, and dispatchResult in struct bagInfo

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1709 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 41 ++++++++++++++++++-
 1 file changed, 40 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index 747a2375..55f07154 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.20";
+   const string interface_version = "1.21";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -204,6 +204,21 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
    };
 
 
+   /**
+    *
+    * A bag can be dispatched without SBD control.
+    * These are the possible modes sent along with IND_PIECE_DELIVER or IND_PIECE_DELIVERED
+    *
+   **/
+   enum operationMode
+   {
+      MODE_AUTO,
+      MODE_MANUAL,
+      MODE_FORCEBAG,
+      MODE_PERFORMANCE_TEST
+   };
+
+
    /**
     *
     * Definition of status of a specific device in the baggage handling system.
@@ -436,6 +451,30 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       **/
       long              baggageWidth;
 
+      /**
+       *
+       * Dispatch notification reference.
+       * Sent conditionally with indications IND_PIECE_DELIVER and IND_PIECE_DELIVERED.
+       *
+      **/
+      string            dispatchReference;
+
+      /**
+       *
+       * Dispatch notification operational mode.
+       * Sent conditionally with indications IND_PIECE_DELIVER and IND_PIECE_DELIVERED.
+       *
+      **/
+      operationMode     dispatchOperationMode;
+
+      /**
+       *
+       * Dispatch notification success code.
+       * Sent conditionally with indications IND_PIECE_DELIVER and IND_PIECE_DELIVERED.
+       *
+      **/
+      errorType         dispatchResult;
+
    };
 
    typedef sequence<octet> image;
-- 
2.41.0.windows.1

